### PR TITLE
reviews/scenarios

### PR DESCRIPTION
**Pull request by gugabs**

### CODE REVIEW

File: src/main/java/cat/udl/eps/softarch/demo/domain/Review.java [hunk 1]
```diff
@@ -16,13 +18,12 @@ public class Review {
     @GeneratedValue(strategy = GenerationType.AUTO)
     private int id;
 
-    @NotBlank
-    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
-    private Date when;
+    private Date when = new Date();
```
Comment (bergacat1) [inline src/main/java/cat/udl/eps/softarch/demo/domain/Review.java:21]: Use ZonedDateTime instead.

File: src/test/resources/features/SubmitReview.feature [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  As a buyer
+  I want to submit a review to a seller
+
+  Background: I'm logged in
+    Given There is no registered user with username "user0"
+    And I'm not logged in
+    When I register a new user with username "user0", email "user@sample.app" and password "password"
+    Then The response code is 201
+    And It has been created a user with username "user0" and email "user@sample.app", the password is not returned
+    And I can login with username "user0" and password "password"
```
Comment (bergacat1) [inline src/test/resources/features/SubmitReview.feature:12]: Background is like a `Given` that will be executed before each test. It's not necessary to do checks with `When` and `Then`

File: src/test/resources/features/SubmitReview.feature [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  Scenario: Submit a review with empty field stars
+    When The buyer submits a new review with username "user0" and message "Great!"
+    Then The response code is 400
+    And The error message is "You must to provide a number of stars in your review"
+    And The review has not been created
+
+  Scenario: Submit a review before having a transaction with user
+    And I'm not logged in
+    When The buyer submits a new review with username "user0", number of stars 5 and message "Great!"
+    Then The response code is 400
```
Comment (bergacat1) [inline src/test/resources/features/SubmitReview.feature:43]: These tests won't pass yet. You could comment them using `#`

**Review by bergacat1 — CHANGES_REQUESTED** [review]

File: tepDefs.result = stepDefs.mockMvc.perform( [hunk 4]
```diff
@@ -0,0 +1,287 @@
+[33mcommit 90933e646c98f9ffd64068e918c2746d5cad73be[m[33m ([m[1;36mHEAD -> [m[1;32mreview[m[33m, [m[1;31morigin/main[m[33m, [m[1;31morigin/HEAD[m[33m, [m[1;32mmain[m[33m)[m
```
Comment (rogargon) [inline tepDefs.result = stepDefs.mockMvc.perform(:1]: This text shouldn't be here!!!

File: .authenticate())) [hunk 5]
```diff
@@ -0,0 +1,187 @@
+
```
Comment (rogargon) [inline .authenticate())):1]: This text shouldn't be here!!!

**Review by rogargon — APPROVED** [review]

**Pull request merged**